### ISSUE TITLE
Add a flag to disable the fieldsCanMerge validation on disjoint types 

### ISSUE THREAD

**Issue by martinbonnin** [apollographql/apollo-kotlin#4320]
Given this schema:

```graphql
  type Query {
    foo: PropertyValue
  }
  union PropertyValue = PropertyValueBoolean | PropertyValueInt

  type PropertyValueBoolean {
    name: String!
    value: Boolean
  }
  type PropertyValueInt {
    name: String!
    value: Int
  }
```

and this query:

```graphql
  query GetFoo {
    foo {
      ...PropertyValue
    }
  }
  fragment PropertyValue on PropertyValue {
    __typename
    ... on PropertyValueBoolean {
        value
    }
    ... on PropertyValueInt {
        value
    }
  }
```

Validation fails with:

```
`value` cannot be merged with `value`: they have different shapes. 
```

This is expected from reading [the spec](https://spec.graphql.org/draft/#sel-GALRFFfCAACGBssG) although it's not clear what the benefit of this validation is on completely disjoint types and it makes the generated models more akward to use. 

Other users have bumped into the same issue in other projects like [here](https://github.com/graphql/graphql-js/issues/53#issuecomment-1077737208)

We could have a Gradle flag that specifically disables this validation:

```kotlin
apollo {
  fieldOnDisjointTypesMustMerge = false
}
```

More context:
* Spec pull request: https://github.com/graphql/graphql-spec/pull/162
* Slack discussion: https://kotlinlang.slack.com/archives/C01A6KM1SBZ/p1659536084099869

**Comment by pt2121** [comment]
Can I give this a try?

**Comment by martinbonnin** [comment]
Very sorry for the delay @pt2121 ! Thanks for looking into this. I'm reviewing the PR now, I'll follow up there.

**Comment by martinbonnin** [comment]
This is now available in 3.6.0
